npm unable to resolve dependency tree

I'm trying to build the main branch, commit version 3a004f0. The build fails with error:

2.596 npm error code ERESOLVE
2.599 npm error ERESOLVE unable to resolve dependency tree
2.600 npm error
2.600 npm error While resolving: vue-mvp@0.0.0
2.600 npm error Found: pinia@2.3.1
2.600 npm error node_modules/pinia
2.600 npm error   pinia@"^2.2.4" from the root project
2.600 npm error
2.600 npm error Could not resolve dependency:
2.600 npm error peerOptional pinia@">=3.0.0" from pinia-plugin-persistedstate@4.3.0
2.600 npm error node_modules/pinia-plugin-persistedstate
2.600 npm error   pinia-plugin-persistedstate@"^4.1.2" from the root project

--- a/client/package.json
+++ b/client/package.json
@@ -21,7 +21,7 @@
     "dotenv": "^16.4.5",
     "jquery": "^3.7.1",
     "pinia": "^2.2.4",
-    "pinia-plugin-persistedstate": "^4.1.2",
+    "pinia-plugin-persistedstate": "4.2.0",
     "vue": "^3.4.15",
     "vue-router": "^4.2.5"
   },

